release notes/v0.4.0.md

```diff
@@ -0,0 +1,171 @@
+xk6-browser v0.4.0 is here! :tada:
+
+In this minor release, we implemented a large portion of a new API that will help you find elements on a page in a more robust way, fixed critical bugs, and made some improvements and code refactoring!
+
+The team is aware that the extension is currently unstable for many users. This stems from the complexity in how [CDP](https://chromedevtools.github.io/devtools-protocol/) events from the browser and from our own internal event-based system are currently handled, as well as our lack of async support in JavaScript APIs, which introduces race conditions and can lead to intermittent failures. We've been delaying tackling these issues since it requires a substantial rearchitecture of core systems, but rest assured, it's currently our top priority and these issues will be addressed by, hopefully, the next release. If you want to follow along the progress, subscribe to issues [#427](https://github.com/grafana/xk6-browser/issues/427) and [#428](https://github.com/grafana/xk6-browser/issues/428).
+
+
+## New features
+
+### Locator API ([#100](https://github.com/grafana/xk6-browser/issues/100))
+
+The Locator API makes it easier to work with dynamically changing elements. Some of the benefits of using it over existing ways to locate an element (e.g. `Page.$()`) include:
+* Helps to write robust tests by finding an element even if the underlying frame navigates.
+* Makes it easier to work with dynamic web pages and SPAs built with Svelte, React, Vue, etc.
+* Organizes and simplifies testing by abstracting pages with the Page Object Model (POM) pattern. [An example](https://github.com/grafana/xk6-browser/blob/v0.4.0/examples/locator_pom.js)
```

Comment:
The Locator API doesn't do this by itself, but makes it easier to use such patterns.

```suggestion
* Enables the use of test abstractions like the Page Object Model (POM) pattern to simplify and organize tests. See [this example](https://github.com/grafana/xk6-browser/blob/v0.4.0/examples/locator_pom.js).
```